bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/MyStromBulbHandler.java

```diff
@@ -0,0 +1,252 @@
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.CHANNEL_SWITCH;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.HSV;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.MONO;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.RGB;
+import static org.openhab.core.library.unit.Units.SECOND;
+import static org.openhab.core.library.unit.Units.WATT;
+
+import java.lang.reflect.Type;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.Fields;
+import org.openhab.core.cache.ExpiringCache;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.HSBType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@link MyStromBulbHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Frederic Chastagnol - Initial contribution
+ */
+@NonNullByDefault
+public class MyStromBulbHandler extends AbstractMyStromHandler {
+
+    private static class MyStromBulbResponse {
+        public boolean on;
+        public String color = "";
+        public String mode = "";
+        public long ramp;
+
+        @Override
+        public String toString() {
+            return "MyStromBulbResponse{" + "on=" + on + ", color='" + color + '\'' + ", mode='" + mode + '\''
+                    + ", ramp=" + ramp + '}';
+        }
+    }
+
+    private static class MyStromDeviceSpecificInfo extends MyStromBulbResponse {
+        public double power;
+    }
+
+    private static final String COMMUNICATION_ERROR = "Error while communicating to the myStrom bulb: ";
+
+    private final Logger logger = LoggerFactory.getLogger(MyStromBulbHandler.class);
+
+    private final ExpiringCache<Map<String, MyStromDeviceSpecificInfo>> cache = new ExpiringCache<>(
+            Duration.ofSeconds(3), this::getReport);
+
+    private PercentType lastBrightness = PercentType.HUNDRED;
+    private DecimalType lastColorTemperature = new DecimalType(10);
+
+    private static final Type TYPE = new TypeToken<HashMap<String, MyStromDeviceSpecificInfo>>() {
```

Comment:
Move up (under the other `private static final`), so constants are kept together. Maybe also rename to `DEVICE_INFO_MAP_TYPE` or something like that.

Reply:
Moved up. But I remove the `COMMUNICATION_ERROR` constant. It was not used since it is in the abstract class. I had not seen that.